lib/token-assert.js

```diff
@@ -128,14 +126,73 @@ TokenAssert.prototype.differentLine = function(options) {
         return;
     }
 
-    if (token.loc.end.line === nextToken.loc.start.line) {
-        nextToken.whitespaceBefore = '\n';
+    this.linesBetween({
+        token: token,
+        nextToken: nextToken,
+        message: options.message || token.value + ' and ' + nextToken.value + ' should be on different lines',
+        atLeast: 0
+    });
+};
+
+/**
+ * Requires tokens to have a certain amount of lines between them.
+ * Set at least one of atLeast or atMost OR set exactly.
+ *
+ * @param {Object} options.token
+ * @param {Object} options.nextToken
+ * @param {Object} [options.message]
+ * @param {Object} [options.atLeast] At least how many lines the tokens are apart
+ * @param {Object} [options.atMost] At most how many lines the tokens are apart
+ * @param {object} [options.exactly] Exactly how many lines the tokens are apart
+ */
+TokenAssert.prototype.linesBetween = function(options) {
+    var token = options.token;
+    var nextToken = options.nextToken;
+
+    if (!token || !nextToken) {
+        return;
+    }
+
+    var atLeast = options.atLeast;
+    var atMost = options.atMost;
+    var exactly = options.exactly;
+
+    if (atLeast === undefined &&
+        atMost === undefined &&
+        exactly === undefined) {
+        throw new Error('You must specify at least one option');
+    }
+
+    if (exactly !== undefined && (atLeast !== undefined || atMost !== undefined)) {
+        throw new Error('You cannot specify atLeast or atMost with exactly');
+    }
+
+    if (atLeast !== undefined && atMost !== undefined && atMost < atLeast) {
+        throw new Error('atLeast and atMost are in conflict');
+    }
+
+    var blankLines = Math.abs(token.loc.end.line - nextToken.loc.start.line) - 1;
```

Comment:
FWIW, I don't think I agree with this.

```javascript
var x = 2;

var y = 3;
```

I think if we asked people how many lines are between those two lines, the answer we would get is 1. However, if we asked how many new line characters there are, I'm sure we'd get two. I do agree though that specifying -1 linesBetween is weird for sameLine, but I believe it makes more sense for everything else.